### DISCUSSION TITLE
Problem with route editing

### DISCUSSION THREAD

**Discussion by tweaks90** [devemux86/cruiser#515] [Q&A] [answered]
Hi

It seems I stumbled across a little bug in one of the latest beta releases 3.0.43-b11.

I tend to adjust the route when I make coffee breaks. What I do is to move the starting point to the current location and remove all shaping points I had already passed. The idea is to get an up to date estimate of the remaining route and to make adjustments as needed.

Here's a reproducible example I came across today documented as a set of screen-shots.

Starting with the initial route as shown in the attached (init.png) ...
![init](https://github.com/devemux86/cruiser/assets/75031892/381720a9-8181-49aa-a82d-01f53d3dc5e4)

... I moved the starting point near shaping point 3 and removed shaping points 2 and 1 afterwards (in this order.) This results in
![result](https://github.com/devemux86/cruiser/assets/75031892/478a0e3b-ce5d-4100-a73e-a40ad2a77e57)

Shaping point 3 was correctly renamed to 1
![zoom_after](https://github.com/devemux86/cruiser/assets/75031892/afeb6f68-df88-4ff7-b0b5-6adc2bf60a17)

I had occasionally also managed to get completely weird routes with direct lines between points. I had to completely delete the whole route and restart cruiser to "fix" this.

best regards,
Tom

**Comment by devemux86** [comment]
@tweaks90 thanks for the report.

You have a route with 4 shaping points. Move start point near point 3. Delete points 1 and 2.
Where exactly is the problem?

Or you mean the route cannot be calculated because of the positions of your moved points?
In this case there is a message on screen and the old route continues to be displayed.

---

This process has many steps and calculations. It can be dome easier with waypoints list:
- Open waypoints list
- Swipe left to remove start and shaping points 1 and 2
- Press OK to calculate a new route

Even better you can split the route at point 3, an offline process _without route calculation_:
- Long press shaping point 3
- Press More, select Split route

**Reply by devemux86** [reply]
If a route cannot be calculated, please provide the route file and the profile options, so we can test it.

**Reply by tweaks90** [reply]
There was no error message indicating that the calculation had failed.

The resulting route does not look correct to me. It leads from point 1 south? and returns without any extra shaping point anywhere. If the calculation had failed point 3 would not have been renamed to 1, would it ??

regards,
Tom

**Reply by devemux86** [reply]
> latest beta releases 3.0.43-b11

Also please try the latest Beta update.

**Reply by devemux86** [reply]
@tweaks90 if you can reproduce the report, please attach the route and the app profile,
so we can make the same test.

**Comment by tweaks90** [comment]
[...]
> 
> This process has many steps and calculations. It can be dome easier with waypoints list:
> 
>     * Open waypoints list
>     * Swipe left to remove start and shaping points 1 and 2
>     * Press OK to calculate a new route
> 
> Even better you can split the route at point 3, an offline process _without route calculation_:
> 
>     * Long press shaping point 3
>     * Press More, select Split route

Thanks. Indeed that sounds simpler. I will try next time.

Cheers,
Tom

**Comment by tweaks90** [comment, marked as answer]
The latest Beta now correctly shows a calculation error. Makes sense now. 

Also, the suggested workflow to split the soute works like a charm. Properly removes the unneeded part.

Thanks for the great Support!

**Reply by devemux86** [reply]
Many things are improved during the Beta period, it was probably a problem that was fixed.

@tweaks90 thanks for the report and if you see anything else, please write to us!
